src/languages/handlebars.js

```diff
@@ -5,19 +5,42 @@ Author: Robin Ward <robin.ward@gmail.com>
 Description: Matcher for Handlebars as well as EmberJS additions.
 Category: template
 */
-function(hljs) {
-  var BUILT_INS = {'builtin-name': 'each in with if else unless bindattr action collection debugger log outlet template unbound view yield lookup'};
-  function MUSTACHE_CONTENTS() {
-    return  {
-      className: 'name',
-      begin: /[a-zA-Z\.-]+/,
+function (hljs) {
+  var BUILT_INS = {'builtin-name': 'each in with if else unless bindattr action collection debugger log outlet template unbound view lookup'};
+
+  var IDENTIFIER_PLAIN_OR_QUOTED = {
+    begin: /("[^"]+"|'[^']+'|\[[^\]]+\]|\w+)/
```

Comment:
Wouldn't back references be a better use here rather than repeating yourself 3-4 times?

Reply:
If at all, only the first two alternatives could be combined using back-referencens.

Non-greedy matching would make it better readable (and it must be `.*`, not `.+`, because `''`, `""` and `[]` are also allowed (in theory):

```js
 /".*?"|'.*?'|\[.*?\]|\w+/
```

But I don't think that this is intuitive:

```js
/(['"]).*?\1|\[.*?\]|\w+/
```
I actually wanted to use: 

```js
  var IDENTIFIER_PLAIN_OR_QUOTED = {
    begin: /\w+/,
    variants: [
      {begin: /"/, end: /"/},
      {begin: /'/, end: /'/},
      {begin: /[/, end: /]/}
    ]
  };
```
but that didn't work. The main begin-pattern `/\w+/` seemed to be ignored in this case.

Reply:
Non-greedy is more readable.

Variants are alternatives to the main block, not "in addition to" or "after"... If you just removed `begin` would the variants not work?